src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultProjectSnapshotManager.cs

```diff
@@ -58,10 +60,16 @@ internal class DefaultProjectSnapshotManager : ProjectSnapshotManagerBase
                 throw new ArgumentNullException(nameof(workspace));
             }
 
+            if (solutionCloseTracker is null)
+            {
+                throw new ArgumentNullException(nameof(solutionCloseTracker));
+            }
+
             _projectSnapshotManagerDispatcher = projectSnapshotManagerDispatcher;
             _errorReporter = errorReporter;
             _triggers = triggers.OrderByDescending(trigger => trigger.InitializePriority).ToArray();
             Workspace = workspace;
+            _solutionCloseTracker = solutionCloseTracker;
```

Comment:
I guess I still saw this as "push" because the information is communicated out via the notification system, but if you think it fits better to write this as a `ProjectSnapshotChangeTrigger` then I'm happy to do it, it just felt a little odd to have a whole trigger that would update a single `bool` field on this class.

I guess it would have made the wire-up a lot easier..

Reply:
I think it makes sense only because it reduces the dependencies of our project snapshot manager & ultimately follows an existing model

Reply:
I'm converting it.. and you're right, this is much nicer.